ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -1317,6 +1324,66 @@ public void shutdown() throws StoreException {
     shutdown(false);
   }
 
+  /**
+   * Gets the log segment metadata files from in-memory data structures
+   * This method returns List of LogSegmentFiles along with its IndexFiles, BloomFilterFiles
+   */
+  @Override
+  public List<LogInfo> getLogSegmentMetadataFiles(boolean includeActiveLogSegment) {
+    List<LogInfo> result = new ArrayList<>();
+    final Timer.Context context = metrics.fileCopyGetMetadataResponse.time();
+
+    List<FileInfo> logSegments = getLogSegments(includeActiveLogSegment);
+    if (null != logSegments) {
+      for (FileInfo fileInfo : logSegments) {
+        LogSegmentName logSegmentName;
+
+        if (fileInfo.getFileName().isEmpty()) {
+          // This happens when the code is running locally and a single LS exists with the name "log_current"
+          logSegmentName = LogSegmentName.fromFilename(LogSegmentName.SINGLE_SEGMENT_LOG_FILE_NAME);
+        } else {
+          logSegmentName = LogSegmentName.fromFilename(fileInfo.getFileName() + LogSegmentName.SUFFIX);
+        }
+        List<FileInfo> indexFiles = getIndexSegmentFilesForLogSegment(dataDir, logSegmentName);
+        List<FileInfo> bloomFiles = getBloomFilterFilesForLogSegment(dataDir, logSegmentName);
+
+        result.add(new LogInfo(fileInfo, indexFiles, bloomFiles));
+      }
+    }
+    context.stop();
+    return result;
+  }
+
+  /**
+   * Get all log segments in the store.
+   * Param includeActiveLogSegment is used to determine if the active log segment should be included in the result.
+   */
+  private List<FileInfo> getLogSegments(boolean includeActiveLogSegment) {
+    return log.getAllLogSegmentNames().stream()
+        .filter(segmentName -> includeActiveLogSegment || !segmentName.equals(log.getActiveSegment().getName()))
+        .map(segmentName -> log.getSegment(segmentName))
+        .map(segment -> new FileInfo(segment.getName().toString(), segment.getView().getFirst().length()))
```

Comment:
every invocation of getView from segment should also couple with a closeView. Since getView and closeView would increment and decrement a refCounter.